tests/abi/abi.c

```diff
@@ -0,0 +1,175 @@
+    printf("    - Offset: -0x%04x\n      Value: 0x%016lx\n",
+           (GENERATED_COUNT - Index) * REGISTER_SIZE,
+           Generated[Index]);
+}
+
+#define CALL_ITERATION_WITH(FUNCTION_NAME) \
+  CONCATENATE(iterationImplOn_, FUNCTION_NAME)
+
+#define GENERATED_BYTES ((GENERATED_COUNT) * sizeof(REGISTER_C_TYPE))
+
+/* clang-format off */
+#define DEFINE_ITERATION_WITH(FUNCTION_NAME)                                 \
+void CALL_ITERATION_WITH(FUNCTION_NAME)(void) {                              \
+  REGISTER_C_TYPE *Generated = (REGISTER_C_TYPE *) malloc(GENERATED_BYTES);  \
+  generate(Generated);                                                       \
+  dump(Generated);                                                           \
+                                                                             \
+  __asm__ __volatile__ (                                                     \
+    /* Save all the registers */                                             \
+    FOR_EACH(PUSH_REGISTER, REGISTER_LIST)                                   \
+    /* Popularize the stack with random data */                              \
+    REPEAT(PUSH_CONSTANT, STACK_VALUE_COUNT, REGISTER_COUNT, FIRST_REGISTER) \
+    /* Popularize all the registers with random data */                      \
+    INDEXED_FOR_EACH(FILL_REGISTER, REGISTER_LIST)                           \
+    /* Call the function */                                                  \
+    CALL_A_FUNCTION(FUNCTION_NAME)                                           \
+    /* Restore the stack */                                                  \
+    RESTORE_STACK(STACK_VALUE_COUNT)                                         \
+    /* Restore all the registers */                                          \
+    REVERSE_FOR_EACH(POP_REGISTER, REGISTER_LIST)                            \
+    :: "X"(Generated)                                                        \
+    : COMMA_SEPARATED_FOR_EACH(STRINGIFY, REGISTER_LIST), "memory"           \
+  );                                                                         \
+                                                                             \
+  free(Generated);                                                           \
+}
+/* clang-format on */
+
+#ifdef __APPLE__
+#define FUNCTION_NAME_PREFIX _
+#else
+#define FUNCTION_NAME_PREFIX
+#endif
+
+#ifdef REG_
+#undef REG_
+#endif
+#define REG_ "%"
+
+#ifdef LOC_
+#undef LOC_
+#endif
+#define LOC_ "%" STRINGIFY(FRAME_POINTER)
+
+/* clang-format off */
+#define NAKED_FUNCTIONS __asm__ (                                          \
+  /* Define the first function */                                          \
+  STRINGIFY(FUNCTION_NAME_PREFIX) "getTheValue:\n"                         \
+  UPDATE_FRAME_POINTER                                                     \
+  ALLOCATE_ON_HEAP(FRAME_POINTER, (GENERATED_COUNT) * (REGISTER_SIZE))     \
```

Comment:
Why do we need to use the heap?

Reply:
Yes, because we're feeling the stack with those values after this. And it's really hard to keep track of where the values come from when the stack changes all the time. When using the heap, those values get generated and just stay at one single point. Only the pointer needs to be kept track of.

Reply:
Heap usage is gone (yay)